### PR TITLE
CMime: Fix mime types memory allocation

### LINKED ISSUE #25396: [Windows] test executable crashes and fails to run
### Describe the bug
Here is a clear and concise description of what the problem is:
Test executable falls to run for windows.
Git bisect found the following PR to cause

```
 c:\Users\brent\source\repos\xbmc>git bisect bad
 a0aa7baea73198e63463e7f931969c8e849a0ffa is the first bad commit
 commit a0aa7baea73198e63463e7f931969c8e849a0ffa
 Author: Arne Morten Kvarving <spiff@kodi.tv>
 Date:   Sun Feb 11 00:44:13 2024 +0100
 
     add video/VideoFileItemClassify.cpp
 
     start by moving CFileItem::IsVideo
```

However I believe newer test cases that have been introduced also are possibly causing crashes of the test executable on windows.

Jenkins was not correctly detecting a failure in the even of no output (ie a crash) from the test executable.

The following PR was made to correctly have Jenkins report such a failure case. https://github.com/xbmc/xbmc/pull/25390

## Expected Behavior
Here is a clear and concise description of what was expected to happen:

Tests to run on windows

## Actual Behavior

Test executable has no output and crashes.

Confirmed test TestVideoFileItemClassify was initial test that started crash, but I believe other tests using unix paths may also be causing crashes, as disabling the Fileitemclassify test source file still has the test executable still fail on tip of master currently.
I used a checkout of a0aa7baea73198e63463e7f931969c8e849a0ffa specifically to locate the specific test mentioned in this issue causing the crash

## Possible Fix

Definitely path related. https://github.com/xbmc/xbmc/blob/501ec1fae0fb86760548018f02702a9caf8ba997/xbmc/video/test/TestVideoFileItemClassify.cpp#L113
```
 const auto video_tests = std::array{
 //    VideoClassifyTest{"/home/user/video.avi", true, "video/avi"},
 //    VideoClassifyTest{"/home/user/video.avi", true, "", 1},
 //    VideoClassifyTest{"/home/user/video.avi", false, "", 2},
 //    VideoClassifyTest{"/home/user/video.avi", false, "", 3},
 //    VideoClassifyTest{"/home/user/video.avi", false, "", 4},
     VideoClassifyTest{"pvr://recordings/tv/1", true},
     VideoClassifyTest{"pvr://123", false},
 //    VideoClassifyTest{"dvd://VIDEO_TS/video_ts.ifo", true},
     VideoClassifyTest{"dvd://1", true},
 //    VideoClassifyTest{"/home/user/video.not", true, "application/ogg"},
 //    VideoClassifyTest{"/home/user/video.not", true, "application/mp4"},
 //    VideoClassifyTest{"/home/user/video.not", true, "application/mxf"},
 };
```
 The only ones that dont cause it to crash are the uncommented ones.

Confirmed Mimetype is not causing a failure, as well as tag and res.

### To Reproduce
1. Build kodi-test on windows. 
2. Execute kodi-test
3.no output

### Debuglog
The debuglog can be found here:
Kodi-test has no debug log. Crash is specifically caused by test case

### Screenshots 
Here are some links or screenshots to help explain the problem:

## Additional context or screenshots (if appropriate)
Here is some additional context or explanation that might help:

### Your Environment
Used Operating system:

 - [ ] Android
 - [ ] iOS
 - [ ] tvOS
 - [ ] Linux
 - [ ] macOS
 - [X] Windows
 - [ ] Windows UWP

 - Operating system version/name:
 - Kodi version: 22 master 501ec1fae0fb86760548018f02702a9caf8ba997

*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

### PR DESCRIPTION
**Pull request by thexai**
## Description
Fix mime types memory allocation issues

## Motivation and context
Try to fix https://github.com/xbmc/xbmc/issues/25396

But i not understand why fails before and why with this not crash but is need change test results...

If test results are consistent in all platforms this means maybe before only crash in Windows but test result where wrong on others platforms? (not crash but not obtained correct mime)

What is the expected result of?

```
    VideoClassifyTest{"/home/user/video.avi", true, "", 2},
    VideoClassifyTest{"/home/user/video.avi", true, "", 3},
    VideoClassifyTest{"/home/user/video.avi", true, "", 4},
```

Could this also have some impact on runtime, not only test?

## How has this been tested?
Test works now in Windows and seems also continue fine on the others platforms.

## What is the effect on users?
?  Not sure at this point if it's only test issue.

## Types of change

- [X] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **Student submission** (PR was done for educational purposes and will be treated as such)
- [ ] **None of the above** (please explain below)

## Checklist:

- [X] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [X] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: xbmc/utils/Mime.h [hunk 1]
```diff
@@ -41,6 +41,504 @@ class CMime
 
 private:
   static bool parseMimeType(const std::string& mimeType, std::string& type, std::string& subtype);
+};
 
-  static const std::map<std::string, std::string> m_mimetypes;
+class CMimeTypes
+{
+public:
```
Comment (ksooo) [inline xbmc/utils/Mime.h:48]: Could this be private? Classes with public members are breaking the concept of classes as instruments of encapsulation. 

Or make CMimeTypes a struct?

Comment (thexai) [reply xbmc/utils/Mime.h:48]: Done

**Comment by neo1973** [comment]
The problem with this solution is that the map containing the mime types is recreated every time. I suggest this instead: https://github.com/neo1973/xbmc/commit/2ef9028efc04d9e7b0f70e37fe55920db9f79f53

**Comment by thexai** [comment]
Tested the solution of neo1973 and is working too. Closing in favor of it.

@neo1973 open a PR then.

**Pull request closed without merging**